src/configure/mod.rs

```diff
@@ -523,143 +659,106 @@ fn get_offline_autoconfig(context: &Context, param: &LoginParam) -> Option<Login
     None
 }
 
-fn try_imap_connections(
-    context: &Context,
-    mut param: &mut LoginParam,
-    was_autoconfig: bool,
-) -> bool {
-    // progress 650 and 660
-    if let Some(res) = try_imap_connection(context, &mut param, was_autoconfig, 0) {
-        return res;
-    }
-    progress!(context, 670);
-    param.server_flags &= !(DC_LP_IMAP_SOCKET_FLAGS);
-    param.server_flags |= DC_LP_IMAP_SOCKET_SSL;
-    param.mail_port = 993;
-
-    if let Some(at) = param.mail_user.find('@') {
-        param.mail_user = param.mail_user.split_at(at).0.to_string();
-    }
-    if let Some(at) = param.send_user.find('@') {
-        param.send_user = param.send_user.split_at(at).0.to_string();
-    }
-    // progress 680 and 690
-    if let Some(res) = try_imap_connection(context, &mut param, was_autoconfig, 1) {
-        res
-    } else {
-        false
-    }
-}
-
-fn try_imap_connection(
-    context: &Context,
-    param: &mut LoginParam,
-    was_autoconfig: bool,
-    variation: usize,
-) -> Option<bool> {
-    if let Some(res) = try_imap_one_param(context, &param) {
-        return Some(res);
-    }
-    if was_autoconfig {
-        return Some(false);
-    }
-    progress!(context, 650 + variation * 30);
-    param.server_flags &= !(DC_LP_IMAP_SOCKET_FLAGS);
-    param.server_flags |= DC_LP_IMAP_SOCKET_STARTTLS;
-    if let Some(res) = try_imap_one_param(context, &param) {
-        return Some(res);
-    }
-
-    progress!(context, 660 + variation * 30);
-    param.mail_port = 143;
-
-    try_imap_one_param(context, &param)
-}
-
-fn try_imap_one_param(context: &Context, param: &LoginParam) -> Option<bool> {
+async fn try_connect(context: &Context, lp: LoginParam, service: Service) -> TryResult {
+    // It is better to use Display for ServerParams here, but it formats data differenly.
     let inf = format!(
-        "imap: {}@{}:{} flags=0x{:x} certificate_checks={}",
-        param.mail_user,
-        param.mail_server,
-        param.mail_port,
-        param.server_flags,
-        param.imap_certificate_checks
+        "{}: {}@{}:{} security={:?} certificate_checks={}",
```

Comment:
I would prefer to use Display trait of ServerParam here, but then message format of the messages should be unified.